### ISSUE TITLE
Bug in BatchNormalization layer

### ISSUE THREAD

**Issue by channingxiao** [keras-team/keras#12953]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  ubunt 14.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13.1
- Keras version:  2.2.4
- Python version:  2.7
- CUDA/cuDNN version:  
- GPU model and memory:  

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  

**Describe the expected behavior**  

**Code to reproduce the issue**  
Provide a reproducible test case that is the bare minimum necessary to generate the problem.  

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.  

There were many issues about the wiered behavior about BN layer. I had the same issue.

I checked the code in Bn layer, and did some experiments. 

Bug:  At normalization.py   line 132
         def call(self, inputs, training=None):
         the parameter " training"  never passed in. it is always None no matter in what case.
         I think during training , this parameter should be True, and False in testing phase.
Here are some proofs:
1. I added some print at line 179,
         if training != None:
            inputs = tf.Print(inputs,["Batch norm training: ", training])
        else:
            inputs = tf.Print(inputs,["Batch norm training is None\n"])
    As wrote above,     "training" is alwaye None, no matter in fit or predict or evaluate.

2.  I changed "def call(self, inputs, training=None):"  at normalization.py   line 132
    to "def call(self, inputs)", and every thing was good, no errers were reported, which means
   there was no place which passing "training" at all, the default parameter "None" were used all the time.

**Comment by jvishnuvardhan** [comment]
@channingxiao can you provide a standalone simple code to reproduce the issue? Thanks!

**Comment by channingxiao** [comment]
use any network with BatchNormalization layers, train with fit_generator, you can reproduce the issue.
Here is a simple net.
def Testnet(img_w, img_h, img_c):
    input_shape = (img_h, img_w, img_c)
    input_data = Input(name='images', shape=input_shape, dtype='float32')
    x = Conv2D(8, (3, 3), strides=(1,1), padding = "same", kernel_initializer='he_normal', name='conv1')(input_data)
    x = BatchNormalization(name='batc1')(x)
    x = Activation('relu', name='actv1')(x)
    x = MaxPooling2D(pool_size=(3,3), strides=(3,3), padding="same", name="pool1")(x)
    model = Model(inputs=input_data, outputs=x)
   return model

**Comment by jtboing** [comment]
Any update on this issue?
